x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java

```diff
@@ -132,6 +135,22 @@ Listener getSearchProgressActionListener() {
         return progressListener;
     }
 
+    @Override
+    protected Supplier<SearchResponseMerger> getSearchResponseMergerSupplier(
```

Comment:
Do we still need a supplier? I find using a supplier to set a local variable slightly hard to read. Would it make sense to move back the creation of the merger where it was, and have a setter and a getter for it on the task? The set and get on SearchTask could be no-op, while AsyncSearchTask tracks it like it already does?

Reply:
Yeah, good idea. I like that. I put the setter/getter on SearchTask. It will be set when CCS MRT=true and false otherwise.

Reply:
Also it wouldn't be a no-op in SearchTask, since synchronous search for MRT=true also uses the SearchResponseMerger. The distinction is between MRT=true and MRT=false (or local-only).